### PR TITLE
[Inductor][CPP] Modularize the CPP GEMM Template

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140786
* #140921
* #137975
* #141554
* __->__ #141006

**Summary**
Move the common template code, which may be reused in subsequent group GEMM templates, into the standalone sub-templates.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/codegen/cpp_gemm_template.py [hunk 1]
```diff
@@ -40,21 +40,7 @@
...
-    {%- set kernel_args = {"X": X, "W": W, "inp": inp, "x_scale": x_scale, "x_zp": x_zp, "w_scale": w_scale, "w_zp": w_zp,} %}
-{%- else %}
-    {%- set kernel_args = {"X": X, "W": W, "inp": inp} %}
-{%- endif %}
-
-extern "C" {{export_declaration}}
-{{kernel.def_kernel(inputs=kernel_args, outputs={"Y": Y}, aliases=aliases)}}
-{
-    {{kernel.maybe_codegen_profile()}}
+GEMM_TEMPLATE_BLOCKS = r"""
```
Comment (jgong5) [inline torch/_inductor/codegen/cpp_gemm_template.py:43]:
```suggestion
GEMM_TEMPLATE_INIT_BLOCKING = r"""
```

File: torch/_inductor/codegen/cpp_gemm_template.py [hunk 2]
```diff
@@ -121,6 +107,77 @@
         Mt_blocks * Nt_blocks * Kt_blocks * {{num_threads}} >= Mr_blocks * Nr_blocks * Kr_blocks,
         "Not all partitions are assigned."
     );
+"""
+
+GEMM_TEMPLATE_MULTI_THREADS_PARAM = r"""
```
Comment (jgong5) [inline torch/_inductor/codegen/cpp_gemm_template.py:112]: using plural is better? same for other var/func names.
```suggestion
GEMM_TEMPLATE_MULTI_THREADS_PARAMS = r"""
```

**Review by jgong5 — APPROVED** [review]
Small nit on naming. Other LGTM.

File: torch/_inductor/codegen/cpp_gemm_template.py (hunk 1 above)
Comment (leslie-fang-intel) [reply torch/_inductor/codegen/cpp_gemm_template.py:43]: Thanks for the comment and changed.

File: torch/_inductor/codegen/cpp_gemm_template.py (hunk 2 above)
Comment (leslie-fang-intel) [reply torch/_inductor/codegen/cpp_gemm_template.py:112]: Thanks for the comment and change to plural format.

**Pull request closed without merging**